packages/edit-widgets/src/components/sidebar/index.js

```diff
@@ -110,84 +182,37 @@ export default function Sidebar() {
 		};
 	}, [] );
 
-	// currentArea, and isGeneralSidebarOpen are intentionally left out from the dependencies,
-	// because we want to run the effect when a block is selected/unselected and not when the sidebar state changes.
-	useEffect( () => {
-		if (
-			hasSelectedNonAreaBlock &&
-			currentArea === WIDGET_AREAS_IDENTIFIER &&
-			isGeneralSidebarOpen
-		) {
-			enableComplementaryArea(
-				'core/edit-widgets',
-				BLOCK_INSPECTOR_IDENTIFIER
-			);
-		}
-		if (
-			! hasSelectedNonAreaBlock &&
-			currentArea === BLOCK_INSPECTOR_IDENTIFIER &&
-			isGeneralSidebarOpen
-		) {
-			enableComplementaryArea(
-				'core/edit-widgets',
-				WIDGET_AREAS_IDENTIFIER
-			);
-		}
-	}, [ hasSelectedNonAreaBlock, enableComplementaryArea ] );
+	const { enableComplementaryArea } = useDispatch( interfaceStore );
 
-	return (
-		<ComplementaryArea
-			className="edit-widgets-sidebar"
-			header={
-				<ul>
-					<li>
-						<ComplementaryAreaTab
-							identifier={ WIDGET_AREAS_IDENTIFIER }
-							label={
-								selectedWidgetAreaBlock
-									? selectedWidgetAreaBlock.attributes.name
-									: __( 'Widget Areas' )
-							}
-							isActive={ currentArea === WIDGET_AREAS_IDENTIFIER }
-						/>
-					</li>
-					<li>
-						<ComplementaryAreaTab
-							identifier={ BLOCK_INSPECTOR_IDENTIFIER }
-							label={ __( 'Block' ) }
-							isActive={
-								currentArea === BLOCK_INSPECTOR_IDENTIFIER
-							}
-						/>
-					</li>
-				</ul>
+	const onTabSelect = useCallback(
+		( newSelectedTabId ) => {
+			if ( !! newSelectedTabId ) {
+				enableComplementaryArea(
+					editWidgetsStore.name,
+					newSelectedTabId
+				);
 			}
-			headerClassName="edit-widgets-sidebar__panel-tabs"
-			/* translators: button label text should, if possible, be under 16 characters. */
-			title={ __( 'Settings' ) }
-			closeLabel={ __( 'Close Settings' ) }
-			scope="core/edit-widgets"
-			identifier={ currentArea }
-			icon={ isRTL() ? drawerLeft : drawerRight }
-			isActiveByDefault={ SIDEBAR_ACTIVE_BY_DEFAULT }
+		},
+		[ enableComplementaryArea ]
+	);
+
+	return (
+		<Tabs
+			// Due to how this component is controlled (via a value from the
+			// `interfaceStore`), when the sidebar closes the currently selected
+			// tab can't be found. This causes the component to continuously reset
+			// the selection to `null` in an infinite loop.Proactively setting
+			// the selected tab to `null` avoids that.selectedTabId={ isGeneralSidebarOpen ? currentArea : null }
```

Comment:
Is this comment relevant in this PR? I can't see the `selectedTabId` prop being passed

Reply:
The comment is relevant, but I'm not sure what happened to my `selectedTabId` prop! We need to to make sure selecting a block on the canvas can effectively control the selected tab in the sidebar. Pushing a fix for this now.